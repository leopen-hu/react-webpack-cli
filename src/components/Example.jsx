import React, { Fragment, useState } from 'react';

export default function Example() {
  const [count, setCount] = useState(0);

  return (
    <Fragment>
      <p> You clicked {count} times </p>
      <button type="button" onClick={() => setCount(count + 1)}>
        Click me
      </button>
    </Fragment>
  );
}
