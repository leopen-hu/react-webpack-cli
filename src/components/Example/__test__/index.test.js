import React from 'react';
import { create } from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import Example from '../index';

describe('Example component', () => {
  test('it matches the snapshot', () => {
    const component = create(<Example />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('count added when clicked', () => {
    const component = create(<Example />);
    const rootInstance = component.root;
    const button = rootInstance.findByType('button');
    const p = rootInstance.findByType('p');
    expect(p.children[1]).toBe('0');
    act(() => {
      button.props.onClick();
    });
    expect(p.children[1]).toBe('1');
  });
});
