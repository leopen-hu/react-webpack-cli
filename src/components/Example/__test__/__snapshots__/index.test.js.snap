// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example component it matches the snapshot 1`] = `
Array [
  <p>
     You clicked 
    0
     times 
  </p>,
  <button
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>,
]
`;
